include/picolibrary/iterator.h

```diff
@@ -136,6 +137,477 @@ struct Iterator_Traits<T *> {
+{
+    return rhs.base() < lhs.base();
+}
+
+/**
+ * \brief Greater than operator.
+ *
+ * \relatedalso picolibrary::Reverse_Iterator
+ *
+ * \tparam Iterator_LHS The type of iterator adapted by the left hand side of the
+ *         comparison.
+ * \tparam Iterator_RHS The type of iterator adapted by the right hand side of the
+ *         comparison.
+ *
+ * \param[in] lhs The left hand side of the comparison.
+ * \param[in] rhs The right hand side of the comparison.
+ *
+ * \return true if lhs is greater than rhs.
+ * \return false if lhs is not greater than rhs.
+ */
+template<typename Iterator_LHS, typename Iterator_RHS>
+constexpr auto operator>( Reverse_Iterator<Iterator_LHS> const & lhs, Reverse_Iterator<Iterator_RHS> const & rhs ) noexcept
+{
+    return lhs.base() < rhs.base();
+}
+
+/**
+ * \brief Less than or equal to operator.
+ *
+ * \relatedalso picolibrary::Reverse_Iterator
+ *
+ * \tparam Iterator_LHS The type of iterator adapted by the left hand side of the
+ *         comparison.
+ * \tparam Iterator_RHS The type of iterator adapted by the right hand side of the
+ *         comparison.
+ *
+ * \param[in] lhs The left hand side of the comparison.
+ * \param[in] rhs The right hand side of the comparison.
+ *
+ * \return true if lhs is less than or equal to rhs.
+ * \return false if lhs is neither less than nor equal to rhs.
+ */
+template<typename Iterator_LHS, typename Iterator_RHS>
+constexpr auto operator<=( Reverse_Iterator<Iterator_LHS> const & lhs, Reverse_Iterator<Iterator_RHS> const & rhs ) noexcept
+{
+    return not( lhs > rhs );
+}
+
+/**
+ * \brief Greater than or equal to operator.
+ *
+ * \relatedalso picolibrary::Reverse_Iterator
+ *
+ * \tparam Iterator_LHS The type of iterator adapted by the left hand side of the
+ *         comparison.
+ * \tparam Iterator_RHS The type of iterator adapted by the right hand side of the
+ *         comparison.
+ *
+ * \param[in] lhs The left hand side of the comparison.
+ * \param[in] rhs The right hand side of the comparison.
+ *
+ * \return true if lhs is greater than or equal to rhs.
+ * \return false if lhs is neither greater than nor equal to rhs.
+ */
+template<typename Iterator_LHS, typename Iterator_RHS>
+constexpr auto operator>=( Reverse_Iterator<Iterator_LHS> const & lhs, Reverse_Iterator<Iterator_RHS> const & rhs ) noexcept
+{
+    return not( lhs < rhs );
+}
+
+/**
+ * Advance a copy of an iterator n positions.
+ *
+ * \tparam Iterator The adapted iterator type.
+ *
+ * \param[in] n The number of positions to advance the iterator.
+ * \param[in] iterator The iterator to advance.
+ *
+ * \return A copy of the iterator that has been advanced n positions.
+ */
+template<typename Iterator>
+constexpr auto operator+( typename Reverse_Iterator<Iterator>::Offset n, Reverse_Iterator<Iterator> const & iterator ) noexcept
+{
+    return Reverse_Iterator{ iterator.base() - n };
+}
+
+/**
+ * \brief Get the distance between two iterators.
```

Comment:
Tie this function to ::picolibrary::Reverse_Iterator's documentation (\relatedalso picolibrary::Reverse_Iterator).